index.test-d.ts

```diff
@@ -17,6 +17,7 @@ app.register(fastifyBasicAuth, {
     expectType<string>(password)
     expectType<FastifyRequest>(req)
     expectType<FastifyReply>(reply)
+    if (Math.random() > 0.5) return new Error()
```

Comment:
should this throw instead?

Reply:
The document is using `return` and it is actually allowed by this line.
https://github.com/fastify/fastify-basic-auth/blob/fdfe2fb3c474257291c475617dcf18025d0669f8/index.js#L24

Reply:
I would prefer two asserts group instead:
- One returning `Promise<Error>`
- One returning `Promise<void>`